server/src/main/java/org/opensearch/gateway/remote/model/RemoteObjectStore.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.model;
+
+import org.opensearch.core.action.ActionListener;
+
+import java.io.IOException;
+
+/**
+ * An interface to read/write an object from/to a remote storage. This interface is agnostic of the remote storage type.
+ *
+ * @param <T> The object type which can be uploaded to or downloaded from remote storage.
+ */
+public interface RemoteObjectStore<T, U extends RemoteObject<T>> {
```

Comment:
Again name is too broad, call it `RemoteClusterStateEntityStore` maybe

Reply:
Same as above. Why do we want to tie the interface to ClusterState ?

Reply:
The apt name would be `RemoteWriteableEntity` instead of `RemoteObject` since this object just really provides a way to customise ser/de

Reply:
Renamed this class to RemoteWritableEntityStore